src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      case action.type === 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete state[keyToRemove];
+        }
+        break;
+
+      case action.type === 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+    }
```

Comment:
```suggestion
    switch (type) {
      case 'clear':
        for (const key in state) {
          delete state[key];
        }
        break;

      case 'removeProperties':
        for (const keyToRemove of action.keysToRemove) {
          delete state[keyToRemove];
        }
        break;

      case 'addProperties':
        Object.assign(state, action.extraData);
        break;
      default:
        // add defaut case - it might be an error
    }
```